packages/app-lib/src/util/rpc.rs

```diff
@@ -0,0 +1,233 @@
+    }
+
+    async fn run(
+        socket: TcpListener,
+        mut message_receiver: mpsc::UnboundedReceiver<RpcMessage>,
+        handlers: HandlerMap,
+        waiting_responses: WaitingResponsesMap,
+    ) -> Result<()> {
+        let (mut socket, _) = socket.accept().await?;
+        loop {
+            tokio::select! {
+                message = message_receiver.recv() => {
+                    let Some(message) = message else {
+                        break;
+                    };
+                    tracing::debug!("Sending RPC message {message:?}");
+                    let json = serde_json::to_vec(&message)?;
+                    socket.write_all(&json).await?;
+                    socket.flush().await?;
+                },
+                message = parse_object_async_reader(&mut socket) => {
+                    let message: RpcMessage = message?;
+                    tracing::debug!("Received RPC message {message:?}");
+                    if let RpcMessageBody::Call { method, args } = message.body {
+                        let response = match handlers.get(method.as_str()) {
+                            Some(handler) => match handler(args).await {
+                                Ok(result) => RpcMessageBody::Respond {
+                                    response: result,
+                                },
+                                Err(e) => RpcMessageBody::Error {
+                                    error: e.to_string(),
+                                },
+                            }
+                            None => RpcMessageBody::Error {
+                                error: format!("Unknown theseus RPC method {method}"),
+                            },
+                        };
+                        let json = serde_json::to_vec(&RpcMessage {
+                            id: message.id,
+                            body: response,
+                        })?;
+                        socket.write_all(&json).await?;
+                        socket.flush().await?;
+                    } else if let Some(sender) = waiting_responses.lock().unwrap().remove(&message.id) {
+                        let _ = sender.send(match message.body {
+                            RpcMessageBody::Respond { response } => Ok(response),
+                            RpcMessageBody::Error { error } => Err(ErrorKind::RpcError(error).into()),
+                            _ => unreachable!(),
+                        });
+                    }
+                },
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Drop for RpcServer {
+    fn drop(&mut self) {
+        self.abort_handle.abort();
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RpcMessage {
+    id: Uuid,
+    #[serde(flatten)]
+    body: RpcMessageBody,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+enum RpcMessageBody {
+    Call {
+        method: String,
+        args: Vec<Value>,
+    },
+    Respond {
```

Comment:
Using the verb respond here sounds off, as the remaining variants are nouns. What about `Reply`?